### PR TITLE
Dop 3567

### PR DESCRIPTION
**Pull request by caesarbell**
### Stories/Links:

DOP-3567

### Current Behavior:

[Prod link](https://www.mongodb.com/docs/search/?q=setProfilingLevel)

Searching in the current prod does not show all results that are not mapped or not find the correct aliases between `/status` and fetching from realm (i.e. `fetchSearchPropertyMapping`)

### Staging Links:

[Stage link](https://docs-mongodb-org-stg.s3.us-east-2.amazonaws.com/871f224b22b05294098027c6ff5a1d548836cac1/master/landing/caesarbell/DOP-3567/search/index.html?q=setProfilingLevel)

We are providing a temporary solution that allows mapping the aliases on the client

### Notes:

In this PR or a follow-up, we want to see if we can order the results slightly better.

### CODE REVIEW

**Review by rayangler — COMMENTED** [review]
Just a few comments on current alias mapping. Please feel free to let me know if my initial observations are incorrect!

Also, I noticed that filtering by docs site wasn't working in the staging link. Is that expected?

**Review by seungpark — COMMENTED** [review]
i'm not sure if this is actually fixing the issue of search not returning this results.

i've checked some documents out in search.documents collection. and it seems like the most recent docs with the searchProperty as `manual-manual, manual-master, manual-upcoming` are from `2022-09-08T00:00:39.000Z`. think there are the most recent docs than these, and there is nothing cleaning up those obsolete files and documents.

i've manually searched some documents in the collection and found:

ObjectId('6376cb2e0f51dc5446253a3b')

image

why is this not being returned in the search? debugging this...

**Comment by seungpark** [comment]
> i'm not sure if this is actually fixing the issue of search not returning this results.
> 
> i've checked some documents out in search.documents collection. and it seems like the most recent docs with the searchProperty as `manual-manual, manual-master, manual-upcoming` are from `2022-09-08T00:00:39.000Z`. think there are the most recent docs than these, and there is nothing cleaning up those obsolete files and documents.
> 
> i've manually searched some documents in the collection and found:
> 
> ObjectId('6376cb2e0f51dc5446253a3b')
> 
> image
> 
> why is this not being returned in the search? debugging this...

seems like we are using this search aggregation 
```
[
  {
    "$search": {
      "compound": {
        "should": [
          {
            "text": {
              "query": ["setprofilinglevel"],
              "path": ["paragraphs", "code.lang", "code.value", "text"]
            }
          },
```
above portion in theory, should return the screenshot document. it contains many instances of `db.setProfilingLevel()` in the **paragraphs** field and also in the **code.value** field.

however, those two fields are being indexed as follows:

```
    "fields": {
      "code": {
        "fields": {
          "lang": [...],
          "value": [
            {
              "dynamic": true,
              "type": "document"
            },
            {
              "analyzer": "lucene.whitespace",
              "searchAnalyzer": "lucene.whitespace",
              "type": "string"
            }
          ]
        },
```

TLDR lucene.whitespace separates and indexes by whitespace and does not lower case the indexed string.

as for **paragraphs** field:

```
      "paragraphs": {
        "analyzer": "lucene.english",
        "searchAnalyzer": "lucene.english",
        "type": "string"
      }
```
according to classic tokenizer, [periods that are not followed by whitespace are kept as part of token](https://solr.apache.org/guide/7_0/tokenizers.html#classic-tokenizer)

so searching by `db.setProfilingLevel` actually returns results: https://www.mongodb.com/docs/search/?q=db.setProfilingLevel

think there is more room for index exploration here than front end fix for these results to populate as desired

File: src/utils/get-temp-mapping.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+// My need to update if aliases change
+//TODO: should remove when root cause is fixed
+export const setAliasesKeys = {
+  'manual-upcoming': 'manual-v6.0',
```
[the remark this answers is not in the document]
Comment (caesarbell) [reply src/utils/get-temp-mapping.js:4]: @rayangler good question and your observation is not totally off. It could be the way I am understanding things. the `upcoming` mapping is to mainly handle the `search-test` DB as `manual-upcoming` is returned as part of the manifest. 

There is still some confusion around how we are to handle outdated branches, but from what I was told, for the temporary solution we will just render everything. Once we move to a ticket that focuses on solving the root cause we will most likely handle in a better way outdated aliases and branches.

File: src/utils/get-temp-mapping.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+  'manual-manual': 'manual-v6.2',
```
[the remark this answers is not in the document]
Comment (caesarbell) [reply src/utils/get-temp-mapping.js:5]: @rayangler good catch, they did say it was going to change. The mapping seems to be actively changing. I will update this to map to `v6.0` and does that mean `upcoming` maps to `v6.2`?

File: src/utils/get-temp-mapping.js [hunk 3]
```diff
@@ -0,0 +1,11 @@
+// My need to update if aliases change
```
[the remark this answers is not in the document]
Comment (caesarbell) [reply src/utils/get-temp-mapping.js:1]: @rayangler good catch on my grammar. I am going to update this.  The benefits of moving this special aliases mapping to Atlas App Service would be? I am mainly asking for learning purposes.

File: src/utils/parse-marian-manifests.js [hunk 4]
```diff
@@ -1,4 +1,9 @@
 import { compareBranchesWithVersionNumbers } from './compare-branches-with-version-numbers';
+import { tempKey } from './get-temp-mapping';
+
+const locatedOutdatedTemporaryFile = (outDatedManifest) => {
```
[the remark this answers is not in the document]
Comment (caesarbell) [reply src/utils/parse-marian-manifests.js:4]: That wasn't the ask for this ticket. This ticket's ask is to render the results from the network call. There will be a separate ticket focusing on fixing the root cause which seems to be a few things ranging from cleaning up stale branches and improving the logic that handles aliases.

File: src/utils/get-temp-mapping.js [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+//TODO: should remove when root cause is fixed
+export const setAliasesKeys = {
```
[the remark this answers is not in the document]
Comment (caesarbell) [reply src/utils/get-temp-mapping.js:3]: @seungpark not sure I understand what the ask is here, can you elaborate?

**Comment by caesarbell** [comment]
> > i'm not sure if this is actually fixing the issue of search not returning this results.
> > i've checked some documents out in search.documents collection. and it seems like the most recent docs with the searchProperty as `manual-manual, manual-master, manual-upcoming` are from `2022-09-08T00:00:39.000Z`. think there are the most recent docs than these, and there is nothing cleaning up those obsolete files and documents.
> > i've manually searched some documents in the collection and found:
> > ObjectId('6376cb2e0f51dc5446253a3b')
> > image
> > why is this not being returned in the search? debugging this...
> 
> seems like we are using this search aggregation
> 
> ```
> [
>   {
>     "$search": {
>       "compound": {
>         "should": [
>           {
>             "text": {
>               "query": ["setprofilinglevel"],
>               "path": ["paragraphs", "code.lang", "code.value", "text"]
>             }
>           },
> ```
> 
> above portion in theory, should return the screenshot document. it contains many instances of `db.setProfilingLevel()` in the **paragraphs** field and also in the **code.value** field.
> 
> however, those two fields are being indexed as follows:
> 
> ```
>     "fields": {
>       "code": {
>         "fields": {
>           "lang": [...],
>           "value": [
>             {
>               "dynamic": true,
>               "type": "document"
>             },
>             {
>               "analyzer": "lucene.whitespace",
>               "searchAnalyzer": "lucene.whitespace",
>               "type": "string"
>             }
>           ]
>         },
> ```
> 
> TLDR lucene.whitespace separates and indexes by whitespace and does not lower case the indexed string.
> 
> as for **paragraphs** field:
> 
> ```
>       "paragraphs": {
>         "analyzer": "lucene.english",
>         "searchAnalyzer": "lucene.english",
>         "type": "string"
>       }
> ```
> 
> according to classic tokenizer, [periods that are not followed by whitespace are kept as part of token](https://solr.apache.org/guide/7_0/tokenizers.html#classic-tokenizer)
> 
> so searching by `db.setProfilingLevel` actually returns results: https://www.mongodb.com/docs/search/?q=db.setProfilingLevel
> 
> think there is more room for index exploration here than front end fix for these results to populate as desired

Good find. I think this is very valuable information and can be used to help us solve the root cause in the next ticket
